File: app/src/main/java/com/yenvth/soilDetectionApp/utils/FileUtils.java
package com.yenvth.soilDetectionApp.utils;

import android.content.Context;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class FileUtils {

    public static String readFile(Context context, String fileName) {
        StringBuilder result = new StringBuilder();
        try {
            InputStream is = context.getAssets().open(fileName);
            BufferedReader reader = new BufferedReader(new InputStreamReader(is, "utf-8"));

            String line;
            while ((line = reader.readLine()) != null) {
                result.append(line);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return result.toString();
    }
}

File: app/src/main/java/com/yenvth/soilDetectionApp/utils/FileUtils.kt
package com.yenvth.soilDetectionApp.utils

import java.lang.StringBuilder
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.IOException
import android.content.Context

object FileUtils {
    fun readFile(context: Context, fileName: String?): String {
        val result = StringBuilder()
        try {
            val `is` = context.assets.open(fileName!!)
            val reader = BufferedReader(InputStreamReader(`is`, "utf-8"))
            var line: String?
            while (reader.readLine().also { line = it } != null) {
                result.append(line)
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return result.toString()
    }
}